### ISSUE TITLE
JSDoc validation for propTypes

### ISSUE THREAD

**Issue by marcodejongh** [yannickcr/eslint-plugin-react#423]
Currently there are no rules for enforcing that propTypes are documented.
In my current project we want to enforce jsdoc on propTypes, because eventually we'll end up with a lot of components with props and it will be hard to tell what is what.

Example:

```js
SomeComponent.propTypes = {
    dispatch: PropTypes.func
};
```

This should give a error something along the lines of "Expected property dispatch to have jsdoc".

Valid would be:

```js
SomeComponent.propTypes = {
    /**
     * Redux action dispatcher function.
     */
    dispatch: PropTypes.func
```

I'm not a 100% sure if it is possible to do. Are PR's welcome for this?

**Comment by lencioni** [comment]
I believe this is possible, but I'm not really convinced of its usefulness. I find that an occasional comment explaining a propType is useful, but most props should have clear names instead of relying on comments to explain them.

It has been a while since you opened this issue. In your project do you still feel like enforcing that every propType has a jsdoc comment on it is the right way to go, or would such a rule just cause a lot of noise?

I think if you want to try this out, you could implement the rule for your team somewhere and run with it for a while. After a while, if you still find it useful and think that others would benefit, I'd be open to reviewing a PR.

I hope this makes sense to you. Let me know if you have any questions! :)

**Comment by marcodejongh** [comment]
We've been using it for a while and I no longer believe that it's very useful to add jsdoc to all props as a rule.
